wireless_modules/middle_module/main.py

```diff
@@ -23,8 +29,11 @@
 my_mq135 = CO2(mq135_pin)
 my_mq135.set_rzero(RZERO)
 
+battery_pin = 33
+battery_reader = BatteryReader(battery_pin, VOLTAGE_FACTOR)
+
 # Set up the wireless module
-middle_module = WirelessModule(MODULE_NUM)
+middle_module = WirelessModule(MODULE_NUM, battery_reader)
 sensors = [my_mpu, my_dht, my_mq135]
 middle_module.add_sensors(sensors)
 
```

Comment:
I know this is bad programming practice, but should we wrap the `middle_module.run()` statement on line 42 (sorry git couldn't let me highlight it) inside a broad `try` and `except` block as a safety net in case something we had no idea of causes our system to fail on the race day. And we could simply reboot the system again, which in most cases should solve the issue (especially for this kind of systems). 
Let me know what you guys think.

Reply:
@hallgchris do you want to implement this in your current PR (Adding the GPS sensor module to back WM) since you made quite a few changes to `main.py` and this would avoid a lot conflicts for whoever merges their PR later.

Reply:
Sure thing!